src/main/java/domain/board/Board.java

```diff
@@ -0,0 +1,153 @@
+package domain.board;
+
+import domain.piece.Piece;
+import domain.piece.PieceType;
+import domain.piece.Team;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Board {
+
+    private final Map<Point, Piece> pieceByPoint;
+    private final PointNodeMapper pointNodeMapper;
+
+    public Board(final Map<Point, Piece> pieceByPoint, final PointNodeMapper pointNodeMapper) {
+        this.pieceByPoint = pieceByPoint;
+        this.pointNodeMapper = pointNodeMapper;
+    }
+
+    public boolean isEnd() {
+        return !isTwoWangsAlive();
+    }
+
+    private boolean isTwoWangsAlive() {
+        return findTeamsOfWang().containsAll(List.of(Team.CHO, Team.HAN));
+    }
+
+    public Team findWinTeam() {
+        if (!isEnd()) {
+            throw new IllegalStateException("아직 게임이 끝나지 않았습니다.");
+        }
+        Set<Team> foundTeam = findTeamsOfWang();
+        if (foundTeam.contains(Team.CHO)) {
+            return Team.CHO;
+        }
+        return Team.HAN;
+    }
+
+    private Set<Team> findTeamsOfWang() {
+        return pieceByPoint.values().stream()
+                .filter(piece -> piece.type() == PieceType.WANG)
+                .map(Piece::team)
+                .collect(Collectors.toSet());
+    }
+
+    public boolean canMove(final Point source, final Point destination) {
+        if (!existsPiece(source)) {
+            return false;
+        }
+        Piece piece = getPieceByPoint(source);
+        return piece.canMove(source, destination, this);
+    }
+
+    public void movePiece(final Point source, final Point destination) {
+        Piece sourcePiece = getPieceByPoint(source);
+        if (!sourcePiece.canMove(source, destination, this)) {
+            throw new IllegalArgumentException(source + " -> " + destination + " [ERROR] 이동할 수 없는 경로입니다.");
+        }
+
+        pieceByPoint.put(destination, sourcePiece);
+        removePiece(source);
+    }
+
+    public boolean existsPiece(final Point point) {
+        if (!existsPoint(point)) {
+            return false;
+        }
+        return pieceByPoint.containsKey(point);
```

Comment:
아래와 같이 변경해볼 수 있을 것 같아요. 😃 

```suggestion
        return existsPoint(point) && pieceByPoint.containsKey(point);
```

Reply:
변경 완료했습니다!